### PR TITLE
Adding ENV, config option to disable agent

### PR DESCRIPTION
**Pull request by blasterpal**
* also primitive/first pass at detecting the type of executing
    environemnt. so agent will be smarter about not running during
    Rake,etc.

### CODE REVIEW

File: lib/app_perf_rpm/introspector.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module AppPerfRpm 
+  class Introspector
+
+    VALID_RUNNERS = %w(Puma Unicorn Sidekiq Sinatra Resque Webbrick)
```
Comment (randy-girard) [inline lib/app_perf_rpm/introspector.rb]: Does Thin register as webbrick? What about passenger? Should we add DelayedJob in here as well? I also use a custom one called Emque. https://github.com/emque/emque-producing and https://github.com/emque/emque-consuming

File: lib/app_perf_rpm/configuration.rb [hunk 2]
```diff
@@ -28,6 +29,7 @@ def reload
         self.sample_rate ||= 10 # Percentage of request to sample
         self.sample_threshold ||= 0 # Minimum amount of duration to sample
         self.dispatch_interval ||= 60 # In seconds
+        self.agent_disabled ||= default_if_blank(ENV["APP_PERF_AGENT_DISABLED"], false )
```
Comment (randy-girard) [inline lib/app_perf_rpm/configuration.rb]: Remove space after false?

File: lib/app_perf_rpm/introspector.rb (hunk 1 above)
Comment (blasterpal) [inline lib/app_perf_rpm/introspector.rb]: This is very incomplete and I'm worried there are a ton of edge cases. I figured I'd get something started and we iterate. I'm testing through it now. 
Use cases I'm focused on:
1. When running rake (assets,etc) it doesn't load the agent at all.
2. Let user declaratively turn off agent.

File: lib/app_perf_rpm/configuration.rb (hunk 2 above)
Comment (blasterpal) [inline lib/app_perf_rpm/configuration.rb]: k

File: lib/app_perf_rpm/introspector.rb (hunk 1 above)
Comment (randy-girard) [inline lib/app_perf_rpm/introspector.rb]: Yup, I agree. We can iterate. Its a good start.

**Comment by blasterpal** [comment]
@randy-girard take a gander.

**Comment by randy-girard** [comment]
I think this is a good start. You want me to pull it in?

**Comment by blasterpal** [comment]
I'm testing out right now for real in my test environment. I'll let you know in a few.

**Comment by randy-girard** [comment]
Hey @blasterpal did you happen to test this out yet?

**Comment by blasterpal** [comment]
I've not tested much yet as I got distracted but I vote let's go ahead and merge. We can always work past if we find issues or revert.

**Pull request merged**